src/daemon/gossip.rs

```diff
@@ -51,20 +52,28 @@ pub async fn gossip_loop(ctx: DaemonContext) -> anyhow::Result<()> {
 }
 
 /// One round of gossip with a particular neighbor.
-async fn gossip_once(ctx: &DaemonContext, conn: &LinkConnection) -> anyhow::Result<()> {
-    fetch_identity(ctx, conn).await?;
-    sign_adjacency(ctx, conn).await?;
-    gossip_graph(ctx, conn).await?;
+async fn gossip_once(
+    ctx: &DaemonContext,
+    neighbor_idpk: IdentityPublic,
+    link_client: Arc<LinkClient>,
+) -> anyhow::Result<()> {
+    log::info!("in gossip_once");
+    fetch_identity(ctx, &neighbor_idpk, link_client.clone()).await?;
+    sign_adjacency(ctx, &neighbor_idpk, link_client.clone()).await?;
+    gossip_graph(ctx, &neighbor_idpk, link_client.clone()).await?;
     Ok(())
 }
 
 // Step 1: Fetch the identity of the neighbor.
-async fn fetch_identity(ctx: &DaemonContext, conn: &LinkConnection) -> anyhow::Result<()> {
-    let remote_fingerprint = conn.remote_idpk().fingerprint();
+async fn fetch_identity(
+    ctx: &DaemonContext,
+    neighbor_idpk: &IdentityPublic,
+    link_client: Arc<LinkClient>,
```

Comment:
Given that all these are passed in from the outside and never cloned/saved elsewhere inside, it would make sense for link_client to be of type &LinkClient rather than Arc<LinkClient>.